ssc/cmod_pvsamv1.cpp

```diff
@@ -2049,135 +2066,138 @@ void cm_pvsamv1::exec( ) throw (general_error)
 
 	for (size_t iyear = 0; iyear < nyears; iyear++)
 	{
-		for (hour = 0; hour < 8760; hour++)
+		//idx is the current array index in the (possibly subhourly) year of weather data or the non-annual array
+		for (size_t inrec = 0; inrec < nrec; inrec++)
 		{
+			idx = inrec + iyear * nrec;
+			if (!wdprov->read(&Irradiance->weatherRecord))
+				throw exec_error("pvsamv1", "could not read data line " + util::to_string((int)(inrec + 1)) + " in weather file");
+
+			size_t hour_of_year = util::hour_of_year(Irradiance->weatherRecord.month, Irradiance->weatherRecord.day, Irradiance->weatherRecord.hour); //this is the index of the hour in the year (0-8759) given the weather file date & timestamp
+
 			// report progress updates to the caller
 			ireport++;
 			if (ireport - ireplast > irepfreq)
 			{
-				percent_complete = percent_baseline + 100.0f *(float)(hour + iyear * 8760) / (float)(insteps);
+				percent_complete = percent_baseline + 100.0f *(float)(inrec + iyear * nrec) / (float)(insteps);
 				if (!update("", percent_complete))
-					throw exec_error("pvsamv1", "simulation canceled at hour " + util::to_string(hour + 1.0) + " in year " + util::to_string((int)iyear + 1) + "in ac loop");
+					throw exec_error("pvsamv1", "simulation canceled at hour " + util::to_string(hour_of_year + 1.0) + " in year " + util::to_string((int)iyear + 1) + "in ac loop");
 				ireplast = ireport;
 			}
 
-			for (size_t jj = 0; jj < step_per_hour; jj++)
-			{
-				double dcPower_kW = PVSystem->p_systemDCPower[idx];
+			double dcPower_kW = PVSystem->p_systemDCPower[idx];
 
-				// Battery replacement
-				if (en_batt && (batt_topology == ChargeController::DC_CONNECTED))
-				{
-					batt->initialize_time(iyear, hour, jj);
-					batt->check_replacement_schedule();
-				}
+			// Battery replacement
+			if (en_batt && (batt_topology == ChargeController::DC_CONNECTED))
+			{
+				// calculate timestep in hour for battery models- this will work for annual simulations
+				// and non-annual simulations are not allowed for battery models, so this should work
+				size_t jj = idx % hour_of_year;
```

Comment:
Oooh good point on checking for zero. I merged patch in last week and got a pretty clean merge, but it will probably take some more work to then merge from patch to develop. Is it better to make another branch before merging into develop, or to wait til this is merged into patch (from a not confusing git perspective)?

Reply:
I think it's fine to merge into a new branch on develop. Once the conflicts are resolved there, there shouldn't be any conflicts merging into develop-- I hope! I just wouldn't merge into patch without fixing the issue first.

Reply:
Will do!